client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -27,6 +31,17 @@ const Launchpad: Step = ( { navigation, flow }: LaunchpadProps ) => {
 	const site = useSite();
 	const launchpadScreenOption = site?.options?.launchpad_screen;
 	const dispatch = useDispatch();
+	const isLoggedIn = useSelector( isUserLoggedIn );
+
+	const fetchingSiteError = useSelect( ( select ) => select( SITE_STORE ).getFetchingSiteError() );
+
+	if ( ! isLoggedIn ) {
+		window.location.replace( `/home/${ siteSlug }` );
+	}
+
+	if ( ! siteSlug || fetchingSiteError?.error === 'unknown_blog' ) {
```

Comment:
`NB`: I'm wondering if there are other site errors that would warrant a redirect from launchpad, and whether or not we'd want to just catch everything?

Reply:
@jeyip -  We can probably just check if fetchingSiteError exists to cover all errors.